### PR TITLE
Capitalize primitives

### PR DESCRIPTION
**Pull request by adamski**
To complete the work for our API requirements, we need to also be able capitalise primitives.

This PR addresses #503

Depends on #502

### CODE REVIEW

File: src/bindgen/mangle.rs [hunk 1]
```diff
@@ -57,11 +59,26 @@ impl<'a> Mangler<'a> {
...
+                        generic.generics(),
+                        last,
+                        Some(self.mangle_separator),
+                        self.capitalize_primitives).mangle();
+
                 self.output.push_str(&sub_path);
             }
             Type::Primitive(ref primitive) => {
-                self.output.push_str(primitive.to_repr_rust());
+                let mut primitive_string = primitive.to_repr_rust().to_owned();
```
Comment (emilio) [inline src/bindgen/mangle.rs:72]: Can we use `Cow::Borrowed`, and only allocate if the new option is used?

Comment (adamski) [reply src/bindgen/mangle.rs]: Done

Comment (adamski) [reply src/bindgen/mangle.rs]: Rebased to master and ran `rustfmt`

**Comment by adamski** [comment]
Closing as we've identified this only partly solves the issue, going to reopen on a new branch titled "PascalCase primitives"

**Pull request closed without merging**